File: Sum_multiples.py
class Solution:
    def sumOfMultiples(self, n: int) -> int:
        if not isinstance(n, int):
            raise TypeError('n must be a valid int type')

        if n <= 0:
            return None

        nums = []

        for i in range(1, n+1):
            if i % 3 == 0 or i % 5 == 0 or i % 7 == 0:
                nums.append(i)

        return sum(nums)


obj = Solution()
print(obj.sumOfMultiples(7))
print(obj.sumOfMultiples(10))
print(obj.sumOfMultiples(9))

File: same_solution_with_different_language/Sum_multiples.js
/**
 * @param {number} n
 * @return {number}
 */
var sumOfMultiples = function (n) {
    if (!Number.isInteger(n)) {
        throw new TypeError("n must be a valid Number");
    } else if (n <= 0) {
        return null;
    }

    let nums = [];

    for (let i = 0; i <= n; i++) {
        if (i % 3 == 0 || i % 5 == 0 || i % 7 == 0) {
            nums.push(i);
        }
    }

    let sum = 0;

    for (let j = 0; j < nums.length; j++) {
        sum += nums[j];
    }
    return sum;
};

console.log(sumOfMultiples(7));
console.log(sumOfMultiples(10));
console.log(sumOfMultiples(9));
